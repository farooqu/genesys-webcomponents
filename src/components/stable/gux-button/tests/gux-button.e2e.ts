import { newSparkE2EPage, a11yCheck } from '../../../../../tests/e2eTestUtils';

describe('gux-button', () => {
  describe('#render', () => {
    [
      {
        clickable: true,
        description: 'should render default button',
        html: '<gux-button>Button</gux-button>'
      },
      {
        clickable: true,
        description: 'should render primary button',
        html: '<gux-button accent="primary">Button</gux-button>'
      },
      {
        clickable: true,
        description: 'should render secondary button',
        html: '<gux-button accent="secondary">Button</gux-button>'
      },
      {
        clickable: true,
        description: 'should render tertiary button',
        html: '<gux-button accent="tertiary">Button</gux-button>'
      },
      {
        clickable: true,
        description: 'should render invalid button',
        html: '<gux-button accent="invalid">Invalid accent</gux-button>'
      },
      {
        clickable: false,
        description: 'should render disabled default button',
        html: '<gux-button disabled>Button</gux-button>'
      },
      {
        clickable: false,
        description: 'should render disabled primary button',
        html: '<gux-button accent="primary" disabled>Button</gux-button>'
      },
      {
        clickable: false,
        description: 'should render disabled secondary button',
        html: '<gux-button accent="secondary" disabled>Button</gux-button>'
      },
      {
        clickable: false,
        description: 'should render disabled tertiary button',
        html: '<gux-button accent="tertiary" disabled>Button</gux-button>'
      },
      {
        clickable: false,
        description: 'should render disabled invalid button',
        html: '<gux-button accent="invalid" disabled>Invalid accent</gux-button>'
      }
    ].forEach(({ description, html, clickable }) => {
      it(description, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-button');
        const onClickSpy = await element.spyOnEvent('click');
        const expectOnclickEvents = clickable ? 1 : 0;

        await a11yCheck(page);
        expect(element.outerHTML).toMatchSnapshot();

        await element.click();
        await page.waitForChanges();

        expect(onClickSpy).toHaveReceivedEventTimes(expectOnclickEvents);
      });
    });
  });

  describe('click', () => {
    it('should fire a click event when an enabled button slot content is clicked', async () => {
      const html = '<gux-button><span>Span</span></gux-button>';
      const page = await newSparkE2EPage({ html });
      const element = await page.find('gux-button');
      const onClickSpy = await element.spyOnEvent('click');
      const span = await page.find('span');

      await span.click();
      await page.waitForChanges();

      expect(onClickSpy).toHaveReceivedEventTimes(1);
    });

    it('should not fire a click event when a disabled button slot content is clicked', async () => {
      const html = '<gux-button disabled><span>Span</span></gux-button>';
      const page = await newSparkE2EPage({ html });
      const element = await page.find('gux-button');
      const onClickSpy = await element.spyOnEvent('click');
      const span = await page.find('span');

      await span.click();
      await page.waitForChanges();

      expect(onClickSpy).toHaveReceivedEventTimes(0);
    });
  });
  describe('focus', () => {
    it('should be programmatically focusable', async () => {
      const html = '<gux-button>Button</gux-button>';
      const page = await newSparkE2EPage({ html });
      const element = await page.find('gux-button');

      await element.callMethod('focus');
      const focusedElement = await page.evaluateHandle(
        () => document.activeElement.nodeName
      );
      expect(await focusedElement.jsonValue()).toBe('GUX-BUTTON');
    });
  });
});
