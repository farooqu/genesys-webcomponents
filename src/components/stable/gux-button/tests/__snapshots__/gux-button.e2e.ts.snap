// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-button #render should render default button 1`] = `"<gux-button class=\\"hydrated\\">Button</gux-button>"`;

exports[`gux-button #render should render disabled default button 1`] = `"<gux-button disabled=\\"\\" class=\\"hydrated\\">Button</gux-button>"`;

exports[`gux-button #render should render disabled invalid button 1`] = `"<gux-button accent=\\"invalid\\" disabled=\\"\\" class=\\"hydrated\\">Invalid accent</gux-button>"`;

exports[`gux-button #render should render disabled primary button 1`] = `"<gux-button accent=\\"primary\\" disabled=\\"\\" class=\\"hydrated\\">Button</gux-button>"`;

exports[`gux-button #render should render disabled secondary button 1`] = `"<gux-button accent=\\"secondary\\" disabled=\\"\\" class=\\"hydrated\\">Button</gux-button>"`;

exports[`gux-button #render should render disabled tertiary button 1`] = `"<gux-button accent=\\"tertiary\\" disabled=\\"\\" class=\\"hydrated\\">Button</gux-button>"`;

exports[`gux-button #render should render invalid button 1`] = `"<gux-button accent=\\"invalid\\" class=\\"hydrated\\">Invalid accent</gux-button>"`;

exports[`gux-button #render should render primary button 1`] = `"<gux-button accent=\\"primary\\" class=\\"hydrated\\">Button</gux-button>"`;

exports[`gux-button #render should render secondary button 1`] = `"<gux-button accent=\\"secondary\\" class=\\"hydrated\\">Button</gux-button>"`;

exports[`gux-button #render should render tertiary button 1`] = `"<gux-button accent=\\"tertiary\\" class=\\"hydrated\\">Button</gux-button>"`;
